### PR TITLE
feat: signed authentication

### PR DESCRIPTION
**Pull request by oed**
This adds support for signed authentication.

### CODE REVIEW

**Comment by oed** [comment]
Perhaps makes sense to bump this package to 1.0.0?

File: src/index.ts [hunk 1]
```diff
@@ -63,21 +63,42 @@ interface DecryptJWEParams {
...
 
+const sign = async (
+  payload: Record<string, any>,
+  did: string,
+  secretKey: Uint8Array,
+  protectedHeader?: Record<string, any>
+) => {
+  const kid = `${did}#${did.split(':')[2]}`
+  const signer = NaclSigner(u8a.toString(secretKey, B64))
+  const header = toStableObject(Object.assign(protectedHeader || {}, { kid, alg: 'EdDSA' }))
```
Comment (PaulLeCam) [inline src/index.ts:79]: Not an issue but we could have instead in the arguments `protectedHeader: Record<string, any> = {}` and then here `toStableObject({ ...protectedHeader, kid, alg: 'EdDSA' })`

File: src/index.ts [hunk 2]
```diff
@@ -63,21 +63,42 @@ interface DecryptJWEParams {
...
-  did_authenticate: ({ did }, params: AuthParams) => {
-    return { did, paths: params.paths }
+  did_authenticate: async ({ did, secretKey }, params: AuthParams) => {
+    const response = await sign(
+      {
+        did,
+        aud: params.aud,
+        nonce: params.nonce,
+        paths: params.paths,
+        exp: Math.floor(Date.now() / 1000) + 600, // expires 10 min from now
```
Comment (PaulLeCam) [inline src/index.ts:91]: Maybe this could be optionally provided in `AuthParams`, in case the consumer may want to change this expiration time?

**Review by PaulLeCam — APPROVED** [review]

File: src/index.ts (hunk 2 above)
Comment (oed) [reply src/index.ts:91]: I don't think that should be up to the consumer.

File: src/index.ts (hunk 1 above)
Comment (oed) [reply src/index.ts:79]: good point!

File: src/index.ts (hunk 2 above)
Comment (PaulLeCam) [reply src/index.ts:91]: There might be different security requirements, no? An OTP system might want to limit the validity to 30 seconds, while an email verification system could leave up to 30 minutes to authenticate for example.

Comment (oed) [reply src/index.ts:91]: Right, I guess this should be addressed in the EIP2844 spec. Let's update it if it becomes a problem.

**Pull request merged**